src/raft_log.rs

```diff
@@ -252,6 +261,13 @@ impl<T: Storage> RaftLog<T> {
                 self.last_index()
             )
         }
+        let entries = self
+            .slice(
```

Comment:
Could this use `unstable_entries` instead of `slice`? If so, the unwrap would become unnecessary here.

Reply:
I worry in this situation that there would be entries written to stable storage, but not yet committed.

You can see here: https://docs.rs/raft/0.4.0/raft/raw_node/struct.Ready.html#structfield.committed_entries

> CommittedEntries specifies entries to be committed to a store/state-machine. These have previously been committed to stable store.

Meanwhile: https://docs.rs/raft/0.4.0/raft/raw_node/struct.Ready.html#structfield.entries

> Entries specifies entries to be saved to stable storage BEFORE Messages are sent.

So if we only rely on `unstable_entries` I think it means we might not see all the uncommitted entries.